plugins/scaffolder-backend/sample-templates/v1beta2-demo/template.yaml

```diff
@@ -70,6 +76,7 @@ spec:
         url: https://github.com/backstage/community/tree/main/backstage-community-sessions
 
     - id: publish
+      if: '{{ not parameters.dryRun }}'
```

Comment:
So what does the `not` return? I think I'm a bit unclear on this one - this is a quoted string, could the result here be `'false'` which evaluates to `true`? :)

Reply:
`not` returns a boolean, but you are right this is not that easy. The boolean is converted to `"false"` in the handlebars template. Afterwards we parse it to JSON so we have `false` again and the `isTruthy` function returns `false` which we evaluate against.

The JSON parsing for `if` works different than for inputs or outputs. For `if` is is always parsed, but input and outputs are only parsed if they start with `"`, `{` or `[`.

Reply:
Alright, the difference in behaviour is what I missed first

Reply:
I actually don't like the difference. But checking for equality with `false`/`true` would be more or less the same 😉